lib/motors/can_messages.cpp

```diff
@@ -0,0 +1,53 @@
+#include "can_messages.hpp"
+
+namespace hyped::motors {
+
+std::vector<uint8_t> convertToBytes(int num, int length)
+{
+  std::vector<uint8_t> bytes = {0};
+
+  for (int i = 0; i < length; i++) {
+    bytes[i] = (num >> 8 * (length - (i + 1))) & 0xFF;
+  }
+  return bytes;
+}
+
+core::Result CanMessages::CanSend(int operation, int location, int data)
+{
+  io::CanFrame frame;
+
+  frame.can_id  = 0x00;
+  frame.can_dlc = 8;
+  frame.data[0] = convertToBytes(operation, 1)[0];
+  frame.data[1] = convertToBytes(location, 2)[0];
+  frame.data[2] = convertToBytes(location, 2)[1];
+  frame.data[3] = 0x00;
+  frame.data[4] = convertToBytes(data, 5)[1];
+  frame.data[5] = convertToBytes(data, 5)[2];
+  frame.data[6] = convertToBytes(data, 5)[3];
+  frame.data[7] = convertToBytes(data, 5)[4];
+
+  core::Result end = can_->send(frame);
+  return end;
+}
+
+core::Result CanMessages::CanError(int error)
+{
+  io::CanFrame frame;
+
+  frame.can_id  = convertToBytes(error, 1)[0];
+  frame.can_dlc = 8;
+  frame.data[0] = 0x00;
+  frame.data[1] = convertToBytes(/*location*/ 0, 2)[0];  // wherever the location is
+  frame.data[2] = convertToBytes(0, 2)[0];
```

Comment:
i don't quite get what's going on here. I think you want to call `convertToBytes(error, 2)` or something then fill `data[0]..[1]` with the indexes of that?

or what do you mean by wherever the location is

Reply:
Oh that was just because since I assumed all messages go to a central computer, the value could be hardcoded as a constant, and so it did not require a variable to handle it - i should have specified that in the code sorry